services/src/modules/build-federated-schema.ts

```diff
@@ -69,8 +73,17 @@ export function buildSchemaFromFederatedTypeDefs({
 
   // Add directives back to the SDL
   const fullTypeDefWithoutDirectives = parse(printSchema(compositionResult.schema));
-  const allDirectivesUsages = mergeDirectivesUsages(serviceDefinitions.map(sd => sd.directivesUsages));
-  const fullTypeDefWithDirectives = addDirectivesToTypedefs(allDirectivesUsages, fullTypeDefWithoutDirectives);
+  const objectTypeDirectives = mergeObjectTypeDirectivesUsages(
```

Comment:
Make sure these handle nulls well i.e. a servicedefinition with object defs and no field defs, etc

Reply:
All variables has default value of `{}`. So it should be ok.

Reply:
Awsome, I just wasn't 100% sure on this 👍